src/pkg/packager/common.go

```diff
@@ -443,6 +444,24 @@ func (p *Packager) validatePackageChecksums() error {
 	return nil
 }
 
+// validatePackageArchitecture validates that the package architecture matches the target cluster architecture.
+func (p *Packager) validatePackageArchitecture() error {
+	// Attempt to connect to a cluster to get the architecture.
+	if cluster, err := cluster.NewClusterWithWait(cluster.DefaultTimeout, false); err == nil {
```

Comment:
```suggestion
	if cluster, err := cluster.NewCluster(); err == nil {
```

We should not wait for a cluster here since this would cause a 30s delay to any Zarf packages that didn't use a cluster or that deployed a cluster themselves.